### DISCUSSION TITLE
Support Offline loading (👍11)

### DISCUSSION THREAD

**Discussion by DirtyF** [11ty/eleventy#3970] [Enhancement Queue]
What would it take for 11ty to offer PWA capabilities by default?

When HTTPS is activated, 11ty could offer:
-  Service Worker for offline support
-  a `manifest.json`

**Comment by nhoizey** [comment]
PWA is a generic (marketing) term regrouping a lot of features, which would you like to have by default?

**Comment by nhoizey** [comment]
I could see a Manifest generated from some elements in `_data/metadata.json`, but some mandatory ones might be missing.

I can't see how a generic Service Worker could be generated through, because there are a lot of different strategies for offline support.

You can create a plugin for one opinionated offline support strategy, but having this in the core could limit opportunities for developers wanting to support other strategies.

(I don't like the PWA/offline plugins currently available for Jekyll, for example)

**Comment by nhoizey** [comment]
@DirtyF maybe a good inspiration for such a plugin: https://github.com/gatsbyjs/gatsby/tree/master/packages/gatsby-plugin-offline

**Comment by nhoizey** [comment]
Well, it's based on sw-precache, which has been replaced by [workbox](https://developers.google.com/web/tools/workbox/).

**Comment by DirtyF** [comment]
FYI https://github.com/lavas-project/jekyll-pwa plugin is based on latest Workbox and is fine as a starter.

**Comment by zachleat** [comment]
I like it! We would love this for https://github.com/NebraskaJS/2018.nejsconf.com but I doubt I could do this by the end of next week 😎

**Comment by zachleat** [comment]
This repository is now using [lodash style issue management](https://twitter.com/samselikoff/status/991395669016436736) for enhancements. This means enhancement issues will now be closed instead of leaving them open.

The enhancement backlog can be found here: https://github.com/11ty/eleventy/issues?utf8=%E2%9C%93&q=label%3Aneeds-votes+sort%3Areactions-%2B1-desc+
